metrics_test.go

```diff
@@ -0,0 +1,526 @@
+// Copyright 2019, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stackdriver
+
+import (
+	"context"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/golang/protobuf/ptypes/timestamp"
+	distributionpb "google.golang.org/genproto/googleapis/api/distribution"
+	googlemetricpb "google.golang.org/genproto/googleapis/api/metric"
+	monitoredrespb "google.golang.org/genproto/googleapis/api/monitoredres"
+	monitoringpb "google.golang.org/genproto/googleapis/monitoring/v3"
+
+	"go.opencensus.io/metric/metricdata"
+	"go.opencensus.io/resource"
+	"time"
+)
+
+var se = &statsExporter{
+	o: Options{ProjectID: "foo"},
+}
+
+func TestMetricResourceToMonitoringResource(t *testing.T) {
+	tests := []struct {
+		in   *resource.Resource
+		want *monitoredrespb.MonitoredResource
+	}{
+		{in: nil, want: &monitoredrespb.MonitoredResource{Type: "global"}},
+		{in: &resource.Resource{}, want: &monitoredrespb.MonitoredResource{Type: "global"}},
+		{
+			in: &resource.Resource{
+				Type: "foo",
+			},
+			want: &monitoredrespb.MonitoredResource{
+				Type: "foo",
+			},
+		},
+		{
+			in: &resource.Resource{
+				Type:   "foo",
+				Labels: map[string]string{},
+			},
+			want: &monitoredrespb.MonitoredResource{
+				Type:   "foo",
+				Labels: map[string]string{},
+			},
+		},
+		{
+			in: &resource.Resource{
+				Type:   "foo",
+				Labels: map[string]string{"a": "A"},
+			},
+			want: &monitoredrespb.MonitoredResource{
+				Type:   "foo",
+				Labels: map[string]string{"a": "A"},
+			},
+		},
+	}
+
+	for i, tt := range tests {
+		got := metricRscToMpbRsc(tt.in)
+		if !reflect.DeepEqual(got, tt.want) {
+			gj, wj := serializeAsJSON(got), serializeAsJSON(tt.want)
+			if gj != wj {
+				t.Errorf("#%d: Unmatched JSON\nGot:\n\t%s\nWant:\n\t%s", i, gj, wj)
+			}
+		}
+	}
+}
+
+func TestMetricToCreateTimeSeriesRequest(t *testing.T) {
+	startTimestamp := &timestamp.Timestamp{
+		Seconds: 1543160298,
+		Nanos:   100000090,
+	}
+	endTimestamp := &timestamp.Timestamp{
+		Seconds: 1543160298,
+		Nanos:   100000997,
+	}
+
+	tests := []struct {
```

Comment:
nit: consider adding a test on built-in metrics.

Reply:
Let me look.